src/shared/tpm2-util.c

```diff
@@ -458,24 +458,161 @@ static int tpm2_make_primary(
         return 0;
 }
 
-void tpm2_pcr_mask_to_selection(uint32_t mask, uint16_t bank, TPML_PCR_SELECTION *ret) {
-        assert(ret);
+/* Utility functions for TPMS_PCR_SELECTION. */
+
+static uint32_t tpm2_tpms_pcr_selection_to_mask(const TPMS_PCR_SELECTION *s) {
+        uint32_t mask = 0;
+        for (unsigned i = 0; i < s->sizeofSelect; i++)
+                SET_FLAG(mask, s->pcrSelect[i] << (i * 8), true);
+        return mask;
+}
+
+static TPMS_PCR_SELECTION tpm2_tpms_pcr_selection_from_mask(uint32_t mask, TPMI_ALG_HASH hash) {
+        TPMS_PCR_SELECTION s = {
+                .hash = hash,
+                .sizeofSelect = TPM2_PCRS_MAX / 8,
+                .pcrSelect = {},
```

Comment:
this line is redundant, unspecified fields in compound initialization default to zero anyway. we generally leave them out hence

Reply:
ack